core/src/main/scala/com/spotify/featran/transformers/IQROutlierRejector.scala

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2017 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.featran.transformers
+
+import com.spotify.featran.FeatureRejection
+import com.twitter.algebird.QTreeAggregator
+
+/**
+  * Transform features by passing them through, and rejecting them if they
+  * fall outside of either 1.5 times the IQR below the first quartile,
+  * or 1.5 times the IQR above the third.
+  *
+  * The IGR or inter quartile range is the range between the first and the third quartiles.
+  *
+  * The bin ranges are chosen using the Algebird's QTree approximate data structure. The precision
+  * of the approximation can be controlled with the `k` parameter.
+  *
+  * Missing values are transformed to zero vectors.
+  *
+  * Values contained in the last bucket will report [[FeatureRejection.Outlier]] as rejection.
+  *
+  * When using aggregated feature summary from a previous session, values outside of previously seen
+  * `[min, max]` will also report [[FeatureRejection.Outlier]] as rejection.
+  */
+object IQROutlierRejector {
+
+  /**
+    * Create a new [[IQROutlierRejector]] instance.
+    *
+    * @param rejectLower whether to reject outlier based on the lower igr bound.
+    * @param rejectUpper whether to reject outlier based on the upper iqr bound.
+    * @param k           precision of the underlying Algebird QTree approximation
+    */
+  def apply(name: String,
+            rejectLower: Boolean = true,
+            rejectUpper: Boolean = true,
+            k: Int = QTreeAggregator.DefaultK)
+  : Transformer[Double, BaseQuantileRejector.B, BaseQuantileRejector.C] =
+    new IQROutlierRejector(name, rejectLower, rejectUpper, k)
+
+}
+
+private class IQROutlierRejector(name: String,
+                          override val rejectLower: Boolean,
+                          override val rejectUpper: Boolean,
+                          override val k: Int)
+  extends QuantileOutlierRejector(name, rejectLower, rejectUpper, 4, k) {
+
+  override def calculateBounds(fq: Double, lq: Double): (Double, Double) = {
+    val iqr = lq - fq
+    val l = fq - (iqr * 1.5)
```

Comment:
haven't seen this factor being tuned, but could be a parameter of the transformer

Reply:
Yeah it should at least be tuneable. What's the reasoning for 1.5 as default?

Reply:
yeah, will make it a param. Seems like it is the most common value used for this technique.
https://math.stackexchange.com/questions/966331/why-john-tukey-set-1-5-iqr-to-detect-outliers-instead-of-1-or-2
```
We certainly CAN use whatever outlier bound we wish to use, but we will have to justify it eventually. In the not-so-recent past, it was typical to expect distributions to be Gaussian. With that assumption, ±1IQR is too exclusive, resulting in too MANY outliers, ±2IQR is too inclusive, resulting in too FEW outliers. ±1.5IQR is easy to remember, and is a reasonable compromise, under assumptions of Gaussianity.
```